src/thor/worker.cc

```diff
@@ -70,10 +70,9 @@ thor_worker_t::thor_worker_t(const boost::property_tree::ptree& config,
       bidir_astar(config.get_child("thor")), bss_astar(config.get_child("thor")),
       multi_modal_astar(config.get_child("thor")), timedep_forward(config.get_child("thor")),
       timedep_reverse(config.get_child("thor")), isochrone_gen(config.get_child("thor")),
-      matcher_factory(config, graph_reader), reader(graph_reader), controller{} {
-  // If we weren't provided with a graph reader make our own
-  if (!reader)
-    reader = matcher_factory.graphreader();
+      reader(graph_reader ? graph_reader
+                          : std::make_shared<baldr::GraphReader>(config.get_child("mjolnir"))),
+      matcher_factory(config, reader), controller{} {
```

Comment:
this code change is good but, you'll want to reorder the members in the header file so that `reader` is declared before `matcher_factor` otherwise some compilers will warn

Reply:
flip these two lines: https://github.com/valhalla/valhalla/blob/master/valhalla/thor/worker.h#L130-L131 so that reader comes first

Reply:
ah  you did that already hahha, nice!

Reply:
yeah my clang did warn ;)